rapid7_insightvm/komand_rapid7_insightvm/actions/get_asset_vulnerabilities/action.py

```diff
@@ -17,8 +19,49 @@ def __init__(self):
     def run(self, params={}):
         resource_helper = ResourceHelper(self.connection.session, self.logger)
         asset_id = params.get(Input.ASSET_ID)
-        endpoint = endpoints.VulnerabilityResult.vulnerabilities_for_asset(self.connection.console_url, asset_id)
+        risk_score = params.get(Input.GET_RISK_SCORE, False)
 
+        endpoint = endpoints.VulnerabilityResult.vulnerabilities_for_asset(self.connection.console_url, asset_id)
         resources = resource_helper.paged_resource_request(endpoint=endpoint,
                                                            method='get')
-        return {Output.VULNERABILITIES: resources}
+        if not risk_score:
+            return {Output.VULNERABILITIES: resources}
+        else:
+            resources = self.get_vulnerabilities(resources)
+            return {Output.VULNERABILITIES: resources}
+
+    async def async_get_vulnerabilities(self, vuln_ids: list) -> list:
+        """
+        Prepares a list of vulnerability ids for asynchronous API calls
+        :param vuln_ids: A list of vulnerability ids
+        :return: A list of vulnerability details
+        """
+        connection = self.connection.async_connection
+        async with connection.get_async_session() as async_session:
+            tasks: [asyncio.Future] = []
+            for vuln_id in vuln_ids:
+                endpoint = endpoints.Vulnerability.vulnerability(self.connection.console_url, vuln_id)
+                tasks.append(asyncio.ensure_future(connection.async_request(session=async_session,
+                                                                            endpoint=endpoint, method='get')))
+            vulnerabilities = await asyncio.gather(*tasks)
+            return vulnerabilities
+
+    def get_vulnerabilities(self, resources: list) -> list:
+        """
+        Finds risk Scores for a list of asset_vulnerabilities at attaches them
+        :param resources: A list of asset_vulnerabilities
+        :return: A list of asset_vulnerabilities with risk Scores
+        """
+        vuln_ids = list()
+        risk_score = list()
+        for resource in resources:
+            vuln_ids.append(resource.get('id'))
+        vulnerabilities = asyncio.run(self.async_get_vulnerabilities(vuln_ids))
```

Comment:
What is going to happen here if `resource.get('id')` returns None? Will it break downstream code? It might be better to try/except, reference by key, and `continue` on KeyError.